src/auth/auth.service.ts

```diff
@@ -134,4 +150,57 @@ export class AuthService {
 
     return { access_token, refresh_token };
   }
+
+  async  googleLogin(id_token: string) {
+    const ticket = await oAuthClient.verifyIdToken({
+      idToken: id_token,
+      audience: process.env.GOOGLE_CLIENT_ID,
+    });
+  
+    const payload = ticket.getPayload();
+    validateGooglePayload(payload);
+  
+    const { email: userEmail, sub: googleId, name: userName } = payload!;
+  
+    let user = await userService.findUserByAuthProvider("google", googleId);
+  
+    if (!user) {
+      user = await prisma.user.findUnique({ where: { email: userEmail! } });
+  
+      if (!user) {
+        user = await userService.createUser(userName!, userEmail!, GOOGLE_PLACEHOLDER) || null;
+
+        if(!user) {
+          throw new UnprocessableEntityError({
```

Comment:
WE probably need to log this case to understand why something went wrong.

Reply:
suggest change